modules/compactor/compactor.go

```diff
@@ -308,10 +318,17 @@ func countSpans(dataEncoding string, objs ...[]byte) int {
 
 		for _, b := range t.Batches {
 			for _, ilm := range b.ScopeSpans {
-				spans += len(ilm.Spans)
+				for _, span := range ilm.Spans {
+					discardedMap[tempoUtil.TraceIDToHexString(span.TraceId)]++
```

Comment:
This code iterates all spans to extract `traceId`.
However, as @mdisibio mentioned in issue #1931. There may be millions spans discarded. therefore, this approach is highly inefficient

Any suggestions for obtaining taceId more efficient?

Reply:
Perhaps just logging the spans here at a debug level instead of returning them?

Reply:
I changed to log number of spans and traceId if number of `ilm.Spans` greater than zero.